### PR TITLE
Generate metadata from commits too.

### PR DESCRIPTION
**Pull request by priyank-p**
This also does not add duplicate PR in LinkParser.
Fixes: https://github.com/nodejs/node-core-utils/issues/156

### CODE REVIEW

**Comment by joyeecheung** [comment]
This could result in duplicate entries in `git-node`...`LandingSession#amend()` needs to remove those from the original commit message during merging.

File: lib/metadata_gen.js [hunk 1]
```diff
@@ -10,11 +10,16 @@ class MetadataGenerator {
...
-    const { owner, repo, pr, reviewers } = data;
+    const { owner, repo, pr, reviewers, commits } = data;
     this.owner = owner;
     this.repo = repo;
     this.pr = pr;
     this.reviewers = reviewers;
+    this.op = pr.bodyHTML;
+
+    commits.forEach(commit => {
+      this.op += commit.commit.messageBodyHTML;
```
Comment (joyeecheung) [inline lib/metadata_gen.js]: If there are multiple commits in one PR, a refs in commit A does not necessarily means it is referenced by commit B...need to figure out how to make this work

Comment (cPhost) [reply lib/metadata_gen.js]: So here i guess you mean some times a PR's commits needs to land in not just one but multiple commits (mostly 2-3 commits) so we want to output them differently, per commit.

Comment (joyeecheung) [reply lib/metadata_gen.js]: @cPhost Yes. Thinking about that, the best we could do is printing these differently for different commits, but don't write them into the actual output consumed by git-node

Comment (cPhost) [reply lib/metadata_gen.js]: > but don't write them into the actual output consumed by git-node

This a part i am bit confused on where exactly to print them (sure not it generated metadata) right.

Comment (joyeecheung) [reply lib/metadata_gen.js]: @cPhost we could do some refactoring with the link parser and get it to return those links separately, say one for the OP, and a map for each commit.

Comment (cPhost) [reply lib/metadata_gen.js]: @joyeecheung if you want to takeover this PR go ahead. I think you will be the right person to solve this. I cannot work on this anymore until after March 10 i got test to prepare for :)

Comment (joyeecheung) [reply lib/metadata_gen.js]: @cPhost Sure, I also plan to write some tests for git-node first so patches can be properly tested.

Good luck on your tests!

Comment (cPhost) [reply lib/metadata_gen.js]: Thanks.

**Comment by codebytere** [comment]
@priyank-p do you ever plan to come back to this or can it be closed?

**Comment by priyank-p** [comment]
I will close this out! I wished I wrote better code back ~2 years ago that I could quickly understand sigh.

**Pull request closed without merging**